### PR TITLE
Add Experiment into the Jupyter notebooks

### PR DESCRIPTION
**Pull request by freud14**

### CODE REVIEW

File: examples/introduction_pytorch_poutyne.ipynb [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "from torchvision import transforms, utils\n",
     "from torchvision.datasets.mnist import MNIST\n",
     "\n",
-    "from poutyne.framework import Model, ModelCheckpoint, CSVLogger, Callback\n",
+    "from poutyne.framework import Model, ModelCheckpoint, CSVLogger, Callback, Experiment\n",
     "from poutyne import torch_to_numpy"
```
Comment (davebulaval) [inline examples/introduction_pytorch_poutyne.ipynb]: We should call the new function in util `set_seed`.

File: examples/introduction_pytorch_poutyne.ipynb [hunk 2]
```diff
@@ -965,6 +900,131 @@
...
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Poutyne Experiment\n",
+    "Most of the time when using Poutyne (or even Pytorch in general), we will find ourselves in an iterative model hyperparameters finetuning loop. For efficient model search, we will usually wish to save our best performing models, their training and testing statistics and even sometimes wish to retrain an already trained model for further tuning. All of the above can be easily implemented with the flexibility of Poutyne Callbacks, but having to define and initialize each and every Callback object we wish for our model quickly feels cumbersome.\n",
+    "\n",
+    "This is why Poutyne also provides an Experiment class, which aims specifically at enabling quick model iteration search, while not sacrifying on the quality of a single experiment - statistics logging, best models saving, etc. Experiment is actually a simple wrapper between a Pytorch module and Poutyne's core Callback objects for logging and saving. Given a working directory where to output the various logging files and a Pytorch module, the Experiment class reduces the whole training loop to a single line.\n",
```
Comment (davebulaval) [inline examples/introduction_pytorch_poutyne.ipynb]: This is why Poutyne provides an Experiment class

File: examples/introduction_pytorch_poutyne.ipynb [hunk 3]
```diff
@@ -965,6 +900,131 @@
...
+    "\n",
+    "The following code uses Poutyne's Experiment class to train a network for 5 epochs. Notice how the code is quite simpler than the code in the Poutyne Callbacks section while doing more. Once trained for 5 epochs, it is then possible to resume the optimization at the 5th epoch for 5 more epochs until the 10th epoch."
```
Comment (davebulaval) [inline examples/introduction_pytorch_poutyne.ipynb]: Also refering to the doc ?

**Comment by davebulaval** [comment]
We might, in all exemple, use the new function `set_seed`.

File: examples/introduction_pytorch_poutyne.ipynb (hunk 3 above)
Comment (freud14) [reply examples/introduction_pytorch_poutyne.ipynb]: Done in cd0cdb9

File: examples/introduction_pytorch_poutyne.ipynb (hunk 2 above)
Comment (freud14) [reply examples/introduction_pytorch_poutyne.ipynb]: Done in cd0cdb9

File: examples/introduction_pytorch_poutyne.ipynb (hunk 1 above)
Comment (freud14) [reply examples/introduction_pytorch_poutyne.ipynb]: Done in cd0cdb9

**Pull request merged**